central/pruning/pruning.go

```diff
@@ -846,31 +856,64 @@ func (g *garbageCollectorImpl) collectAlerts(config *storage.PrivateConfig) {
 			AddExactMatches(search.ViolationState, storage.ViolationState_ATTEMPTED.String()).
 			AddDays(search.ViolationTime, int64(pruneAttemptedRuntimeAfter)).
 			ProtoQuery()
-		queries = append(queries, q)
+		queryMap[pruneAttemptedRuntimeAfterKey] = q
 	}
 
-	if len(queries) == 0 {
+	if len(queryMap) == 0 {
 		log.Info("No alert retention configuration, skipping")
 		return
 	}
 
 	ctx, cancel := context.WithTimeout(pruningCtx, alertQueryTimeout)
 	defer cancel()
 
-	alertResults, err := g.alerts.Search(ctx, search.DisjunctionQuery(queries...))
-	if err != nil {
-		log.Error(err)
-		return
-	}
+	prunedAlertCount := 0
+	// Go through the various queries and remove those batches.
+	for key, query := range queryMap {
+		log.Infof("Pruning for key %q", key)
+		alertResults, err := g.alerts.Search(ctx, query)
```

Comment:
Since there is already a WalkByQuery below, is this search needed?

Reply:
ah nice catch on that left over remnant.